include/circt/Dialect/HW/HWTypesImpl.td

```diff
@@ -115,6 +115,7 @@ def StructTypeImpl : HWType<"Struct"> {
     using FieldInfo = ::circt::hw::detail::FieldInfo;
     mlir::Type getFieldType(mlir::StringRef fieldName);
     void getInnerTypes(mlir::SmallVectorImpl<mlir::Type>&);
+    llvm::Optional<unsigned> getFieldIndex(mlir::StringRef fieldName);
```

Comment:
I think you can use StringAttr instead of StringRef.

Reply:
We probably want both, but since the existing API here is StringRef for other functions, I wouldn't block this on cleaning that up.  Maybe file an good-first-issue issue?